### PR TITLE
Change rename/inline-rename to no longer marshal symbols from OOP to the host when operating.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
The Rename *API* offers a few flexible ways to work with it to support many features within roslyn.  For example, 'Inline rename' works in a mode where it first determines the set of locations that need to be updated, and then continually computes/cancels a step to determine if renaming those locations would cause conflicts *given* the current name the user has typed so far.

In other words, the flow of information looks like this:

```mermaid
sequenceDiagram
    Host->>OOP: Get rename replacement "locations"
    OOP-->>Host: Here are the "heavyweight locations" to replace.
    Host->>OOP: Cool. Give me the conflicts for these "heavyweight locations" for new name 'Foo'
    OOP->>OOP-Rename: Give me the conflicts for those "heavyweight locations" for new name 'Foo'
    OOP-Rename-->>OOP: Here are the conflicts for 'Foo'
    OOP-->>Host: Here are the conflicts for 'Foo'
    Host->>OOP: How about the conflicts for these "heavyweight locations" for 'Foobar'?
    OOP->>OOP-Rename: Give me the conflicts for those heavyweight locations for new name 'Foobar'
    OOP-Rename-->>OOP: Here are the conflicts for 'Foobar'
    OOP-->>Host: Here's the conflict for 'Foobar'
    Host->>OOP: Rinse/Repeat
```

Currently, this is very expensive as the "heavyweight locations" information contains symbolic information, which then must be hydrated on the host side.  Hydration is expensive as we may have to create compilations locally and lookup symbol keys within them.  Creating compilations can require emitting dlls, running source generators, and so forth.  Each time the host and OOP exchange this info, that expensive work may happen.

The new approach changes things so that instead of passing a very expensive object back and forth, we only pass a lightweight one that does not cause symbol resolution to happen on the host side.  That object is pure trivial data, and can be easily passed back and forth without causing any local compilations to be created.

After looks like this:

```mermaid
sequenceDiagram
    Host->>OOP: Get rename replacement "locations"
    OOP-->>Host: Here are the "lightweight locations" to replace.
    Host->>OOP: Cool. Give me the conflicts for these "lightweight locations" for new name 'Foo'
    OOP->>OOP: Converts the lightweight to heavyweight
    OOP->>OOP-Rename: Give me the conflicts for those "heavyweight locations" for new name 'Foo'
    OOP-Rename-->>OOP: Here are the conflicts for 'Foo'
    OOP-->>Host: Here are the conflicts for 'Foo'
    Host->>OOP: How about the conflicts for these "lightweight locations" for 'Foobar'?
    OOP->>OOP: Converts the lightweight to heavyweight
    OOP->>OOP-Rename: Give me the conflicts for those heavyweight locations for new name 'Foobar'
    OOP-Rename-->>OOP: Here are the conflicts for 'Foobar'
    OOP-->>Host: Here's the conflict for 'Foobar'
    Host->>OOP: Rinse/Repeat
```

While this still has a conversion from lightweight->heavyweight, it is done in-process on the OOP side, which already has the solution and has already built the compilations (which it needed in order to pass the data back to the host in the first place).  There is now no more work done on the host side to support this.

### CODE REVIEW

File: src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.InlineRenameLocationSet.cs [hunk 1]
```diff
@@ -7,22 +7,21 @@
...
+using Microsoft.CodeAnalysis.Rename.ConflictEngine;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.InlineRename
 {
     internal abstract partial class AbstractEditorInlineRenameService
     {
         private class InlineRenameLocationSet : IInlineRenameLocationSet
         {
-            private readonly RenameLocations _renameLocationSet;
+            private readonly LightweightRenameLocations _renameLocationSet;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.InlineRenameLocationSet.cs:19]: inline rename now only holds onto "LightweightRenameLocations" so it can cheaply call back into OOP and so no local symbols/compilations are created for it.

File: src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.SymbolRenameInfo.cs [hunk 2]
```diff
@@ -71,7 +71,7 @@ private partial class SymbolInlineRenameInfo : IInlineRenameInfoWithFileRename
                 _fallbackOptions = fallbackOptions;
                 this.RenameSymbol = renameSymbol;
 
-                this.HasOverloads = RenameLocations.GetOverloadedSymbols(this.RenameSymbol).Any();
+                this.HasOverloads = RenameUtilities.GetOverloadedSymbols(this.RenameSymbol).Any();
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.SymbolRenameInfo.cs:74]: we had static helper methods scattered around 3 places.  If the static method was called in at least 2 distinct locations, i moved to RenameUtilities since htat already exists and seems more conducive to that purpose.

File: src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs [hunk 3]
```diff
@@ -76,7 +76,7 @@ public async Task<IInlineRenameInfo> GetRenameInfoAsync(Document document, int p
             // If rename is invoked on a member group reference in a nameof expression, then the
             // RenameOverloads option should be forced on.
             var forceRenameOverloads = tokenRenameInfo.IsMemberGroup;
-            var symbol = await RenameLocations.ReferenceProcessing.TryGetRenamableSymbolAsync(document, triggerToken.SpanStart, cancellationToken: cancellationToken).ConfigureAwait(false);
+            var symbol = await RenameUtilities.TryGetRenamableSymbolAsync(document, triggerToken.SpanStart, cancellationToken: cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs:79]: especially odd was a nested static type with helpers taht were used all over roslyn.

File: src/Features/Core/Portable/EncapsulateField/AbstractEncapsulateFieldService.cs [hunk 4]
```diff
@@ -310,30 +311,37 @@ private async Task<Solution> EncapsulateFieldsInCurrentProcessAsync(Document doc
...
                 RenameFile: false);
 
-            var initialLocations = await Renamer.FindRenameLocationsAsync(
+            var initialLocations = await Renamer.FindLightweightRenameLocationsAsync(
                 solution, field, options, fallbackOptions, cancellationToken).ConfigureAwait(false);
 
-            var resolution = await initialLocations.Filter(filter).ResolveConflictsAsync(
-                finalName, nonConflictSymbols: null, cancellationToken).ConfigureAwait(false);
+            var resolution = await ConflictResolver.ResolveLightweightConflictsAsync(
+                initialLocations.Filter(filter), finalName, nonConflictSymbols: null, cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/EncapsulateField/AbstractEncapsulateFieldService.cs:318]: "encapsulate field" and "use auto prop" are examples of two other features that engage with the rename engine in a two-phase fashoin (getting the locations to rename, adapting them, and then asking for the rename to be done).

Previously, this would force all the renamed symbols to be marshalled back to the host, now taht no longer happens.

File: src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs [hunk 5]
```diff
@@ -30,7 +30,7 @@ internal static partial class ConflictResolver
         private class Session
         {
             // Set of All Locations that will be renamed (does not include non-reference locations that need to be checked for conflicts)
-            private readonly RenameLocations _renameLocationSet;
+            private readonly HeavyweightRenameLocations _renameLocationSet;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs:33]: this is the component that actually does conflit resolution.  It lives in OOP and actually wants to work with and process symbols/compilations/etc.  as such, it holds onto a HeavyweightRenameLocations instance.

File: src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.cs [hunk 6]
```diff
@@ -68,32 +79,40 @@ internal static partial class ConflictResolver
...
             }
 
-            return await ResolveConflictsInCurrentProcessAsync(
-                renameLocationSet, replacementText, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
+            var heavyweightLocations = await lightweightRenameLocations.ToHeavyweightAsync(cancellationToken).ConfigureAwait(false);
+            if (heavyweightLocations is null)
+                return new ConflictResolution(WorkspacesResources.Failed_to_resolve_rename_conflicts);
+
+            return await ResolveHeavyweightConflictsInCurrentProcessAsync(
+                heavyweightLocations, replacementText, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.cs:87]: once we've marshalled over to OOP, we convert hte lightweight information back to hte heavyweight information needed to do the processing.  This should be cheap on the OOP side as the solutino/compilaitons are already synced and available there.

File: src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.ReferenceProcessing.cs [hunk 7]
```diff
@@ -14,111 +16,17 @@
...
+    internal sealed partial class HeavyweightRenameLocations
     {
         internal static class ReferenceProcessing
         {
-            /// <summary>
-            /// Given a symbol in a document, returns the "right" symbol that should be renamed in
-            /// the case the name binds to things like aliases _and_ the underlying type at once.
-            /// </summary>
-            public static async Task<ISymbol?> TryGetRenamableSymbolAsync(
-                Document document, int position, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.ReferenceProcessing.cs:36]: all these nested static methods were used in many locations in roslyn.  They moved to RenameUtilities as it really didn't make sense to update them to say HeavyweightRenameLocations.ReferenceProcessing.DoSomething.  The helpers have nothing to do with HeavyweightRenameLocations at all.

File: src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs [hunk 8]
```diff
@@ -14,17 +14,17 @@
...
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Rename
 {
     /// <summary>
-    /// Holds the Locations of a symbol that should be renamed, along with the symbol and Solution
-    /// for the set.
+    /// Holds the Locations of a symbol that should be renamed, along with the symbol and Solution for the set. It is
+    /// considered 'heavy weight' because it holds onto large entities (like Symbols) and thus should not be marshaled
+    /// to/from a host to OOP.
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs:25]: note: no marshalling code actually exists for this type.  this is intentional to prevent that from accidentally happening.

File: src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs [hunk 9]
```diff
@@ -48,74 +48,22 @@ internal sealed partial class RenameLocations
...
-            ISymbol symbol,
-            Solution solution,
-            ImmutableArray<ISymbol> referencedSymbols,
-            ImmutableArray<ReferenceLocation> implicitLocations,
-            SymbolRenameOptions options,
-            CodeCleanupOptionsProvider fallbackOptions)
-        {
-            return new RenameLocations(
-                symbol, solution, options, fallbackOptions, locations, implicitLocations, referencedSymbols);
-        }
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs:67]: no point in a factory method that just forwarded thnigs along.  NOte: the order of parameters was different too.  So you'll see several reference to this become a call to the construct, with some reordering.

File: src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs [hunk 10]
```diff
@@ -48,74 +48,22 @@ internal sealed partial class RenameLocations
...
-
         /// <summary>
-        /// Find the locations that need to be renamed.
+        /// Attempts to find all the locations to rename.  Will not cross any process boundaries to do this.
         /// </summary>
-        public static async Task<RenameLocations> FindLocationsAsync(
-            ISymbol symbol, Solution solution, SymbolRenameOptions options, CodeCleanupOptionsProvider fallbackOptions, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs:73]: moved to LightweightRenameLocations

File: src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs [hunk 11]
```diff
@@ -169,30 +117,12 @@ internal sealed partial class RenameLocations
...
-                    foreach (var member in containingType.GetMembers())
-                    {
-                        if (string.Equals(member.MetadataName, symbol.MetadataName, StringComparison.Ordinal) && member is IMethodSymbol && !member.Equals(symbol))
-                        {
-                            yield return member;
-                        }
-                    }
-                }
-            }
-        }
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations.cs:194]: moved to RenameUtilities.

**Comment by CyrusNajmabadi** [comment]
@Cosifne  PTAL :)

File: src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.InlineRenameLocationSet.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.InlineRenameLocationSet.cs:19]: this part of inlinerename is on the host side, so we only want it to hold onto lightweight stuff.

**Comment by ryzngard** [comment]
Looking at this today, love the detailed explanation in the description!

File: src/Features/Core/Portable/EncapsulateField/AbstractEncapsulateFieldService.cs [hunk 12]
```diff
@@ -265,26 +266,26 @@ private async Task<Solution> EncapsulateFieldsInCurrentProcessAsync(Document doc
             if (field.IsReadOnly)
             {
                 // Inside the constructor we want to rename references the field to the final field name.
-                var constructorLocations = GetConstructorLocations(field.ContainingType);
+                var constructorLocations = GetConstructorLocations(solution, field.ContainingType);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/EncapsulateField/AbstractEncapsulateFieldService.cs:269]: previously, we were operating on data that used the Location type.  However, this type references a SyntaxTree, which is not something we marshall to/from oop.  Now it uses DocumentId/TextSpan to match the data we have in this lightweight type.

File: src/Workspaces/Core/Portable/Rename/IRemoteRenamerService.cs [hunk 13]
```diff
@@ -204,7 +183,7 @@ internal sealed class SerializableRenameLocations
         // We use arrays so we can represent default immutable arrays.
 
         [DataMember(Order = 1)]
-        public SerializableRenameLocation[]? Locations;
+        public SerializableRenameLocation[] Locations;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/IRemoteRenamerService.cs:186]: this was never actually possible to be null.

File: src/Workspaces/Core/Portable/Rename/RenameUtilities.cs [hunk 14]
```diff
@@ -175,5 +182,205 @@ private static bool ShouldRenameOnlyAffectDeclaringProject(ISymbol symbol)
 
             return TokenRenameInfo.NoSymbolsTokenInfo;
         }
+
+        public static IEnumerable<ISymbol> GetOverloadedSymbols(ISymbol symbol)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/RenameUtilities.cs:186]: this file is nothing but moves.  no logic changes at all.

File: src/Workspaces/Core/Portable/Rename/Renamer.cs [hunk 15]
```diff
@@ -213,8 +215,11 @@ internal static Task<RenameLocations> FindRenameLocationsAsync(Solution solution
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var renameLocations = await FindRenameLocationsAsync(solution, symbol, options, cleanupOptions, cancellationToken).ConfigureAwait(false);
-            return await renameLocations.ResolveConflictsAsync(newName, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
+            // Since we know we're in the oop process, we know we won't need to make more OOP calls.  Since this is the
+            // rename entry-point that does the entire rename, we can directly use the heavyweight RenameLocations type,
+            // without having to go through any intermediary LightweightTypes.
+            var renameLocations = await HeavyweightRenameLocations.FindLocationsInCurrentProcessAsync(symbol, solution, options, cleanupOptions, cancellationToken).ConfigureAwait(false);
+            return await ConflictResolver.ResolveHeavyweightConflictsInCurrentProcessAsync(renameLocations, newName, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/Renamer.cs:222]: note: this is a nice optimization we can make. this is the entrypoint that says "do the full rename".  because it only needs to send the final results back to the host, we can skip anything related to the LightweightRenameLocations work as that intermediary value is never sent back to the host.

File: src/Features/Core/Portable/EncapsulateField/AbstractEncapsulateFieldService.cs [hunk 16]
```diff
@@ -321,19 +322,26 @@ private async Task<Solution> EncapsulateFieldsInCurrentProcessAsync(Document doc
...
 
             return false;
         }
 
-        private ISet<Location> GetConstructorLocations(INamedTypeSymbol containingType)
-            => GetConstructorNodes(containingType).Select(n => n.GetLocation()).ToSet();
+        private ISet<(DocumentId documentId, TextSpan span)> GetConstructorLocations(Solution solution, INamedTypeSymbol containingType)
+            => GetConstructorNodes(containingType).Select(n =>
+            {
+                var location = n.GetLocation();
```
Comment (ryzngard) [inline src/Features/Core/Portable/EncapsulateField/AbstractEncapsulateFieldService.cs:342]: Not used?

**Comment by Cosifne** [comment]
> @Cosifne PTAL :)

It's on my to-do list :)

File: src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.cs [hunk 17]
```diff
@@ -37,8 +37,19 @@ internal static partial class ConflictResolver
 
         private const string s_metadataNameSeparators = " .,:<`>()\r\n";
 
-        internal static async Task<ConflictResolution> ResolveConflictsAsync(
-            RenameLocations renameLocationSet,
+        /// <summary>
+        /// Performs the renaming of the symbol in the solution, identifies renaming conflicts and automatically
+        /// resolves them where possible.
+        /// </summary>
+        /// <param name="replacementText">The new name of the identifier</param>
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.cs:44]: Do we need to replace ISymbol usage here for `nonConflictSymbols` as well? Since this is already moved over to the LightweightRenameLocations

File: src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    /// <summary>
+    /// Equivalent to <see cref="HeavyweightRenameLocations"/> except that references to symbols are kept in a lightweight fashion
+    /// to avoid expensive rehydration steps as a host and OOP communicate.
+    /// </summary>
+    internal sealed partial class LightweightRenameLocations
+    {
+        public readonly ISymbol Symbol;
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:26]: Are we worried about hydration of this symbol?

File: src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs [hunk 19]
```diff
@@ -0,0 +1,143 @@
...
+               referencedSymbols);
+        }
+
+        /// <summary>
+        /// Find the locations that need to be renamed.  Can cross process boundaries efficiently to do this.
+        /// </summary>
+        public static async Task<LightweightRenameLocations> FindRenameLocationsAsync(
+            ISymbol symbol, Solution solution, SymbolRenameOptions options, CodeCleanupOptionsProvider fallbackOptions, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfNull(solution);
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:88]: Why are these needed for internal nullable enabled code?

File: src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+                    }
+                }
+            }
+
+            // Couldn't effectively search in OOP. Perform the search in-proc.
+            var renameLocations = await HeavyweightRenameLocations.FindLocationsInCurrentProcessAsync(
+                symbol, solution, options, fallbackOptions, cancellationToken).ConfigureAwait(false);
+
+            return new LightweightRenameLocations(
+                symbol, solution, options, fallbackOptions, renameLocations.Locations,
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:125]: Should we dehydrate the symbol here?

**Review by ryzngard — APPROVED** [review]
Overall love this approach. Thanks for the clear explanation of the change and good use of comments to help understand when light/heavy is used. Minor comments/questions

File: src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations1.cs [hunk 21]
```diff
@@ -8,9 +8,9 @@
 
 namespace Microsoft.CodeAnalysis.Rename
 {
-    internal sealed partial class RenameLocations
+    internal sealed partial class HeavyweightRenameLocations
```
Comment (tmat) [inline src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations1.cs:11]: >HeavyweightRenameLocations [](http://example.com/codeflow?start=34&length=26)

Would `SymbolicRenameLocations` or `SemanticRenameLocations`  be better name?

File: src/Workspaces/Core/Portable/Rename/IRemoteRenamerService.cs [hunk 22]
```diff
@@ -204,7 +183,7 @@ internal sealed class SerializableRenameLocations
         // We use arrays so we can represent default immutable arrays.
```
Comment (tmat) [inline src/Workspaces/Core/Portable/Rename/IRemoteRenamerService.cs:183]: >// We use arrays so we can represent default immutable arrays [](http://example.com/codeflow?start=8&length=61)

This is odd. Perhaps it was a workaround for old version of JSON serializer that had a bug?

File: src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.InlineRenameLocationSet.cs (hunk 1 above)
Comment (tmat) [reply src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.InlineRenameLocationSet.cs:19]: LightweightRenameLocations still holds on ISymbol. Why is that?

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.InlineRenameLocationSet.cs:19]: for ease of making this PR.  it doesn't ever actually *marshall* that symbol.  it's just a pointer *on the host* to the original symbol that hte host determined it wants to rename.

File: src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs (hunk 18 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:26]: it's technically not hydrated or involved in marshalling at all.  but i can see how this is confusing.  I can do an immediate followup to remove this.  It will require more passing around of data as parameters.  but htat's likely a good thing.

File: src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations1.cs (hunk 21 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations1.cs:11]: yup.  i like that.

File: src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs (hunk 20 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:125]: will do followup to make this clearer.

File: src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs (hunk 19 above)
Comment (Cosifne) [reply src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:88]: I guess this method is just a copy and paste from some old utilities : )

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:88]: yes.  lots of rename is not NRT enabled.  so we're in a mix of patterns here.  it's not great.

File: src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs (hunk 20 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:125]: followup here: https://github.com/dotnet/roslyn/pull/62803

File: src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations1.cs (hunk 21 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/HeavyweightRenameLocations1.cs:11]: done in followup: https://github.com/dotnet/roslyn/pull/62803

File: src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs (hunk 18 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/LightweightRenameLocations.cs:26]: done in followup: https://github.com/dotnet/roslyn/pull/62803
